lein-cljfmt/src/leiningen/cljfmt.clj

```diff
@@ -30,7 +33,8 @@
         options (-> (:cljfmt project)
                     (dissoc :paths)
                     (assoc :project-root (:root project))
-                    (merge-default-options))]
+                    (merge-default-options)
+                    (merge-file-options))]
```

Comment:
These two lines are in the wrong order; the way it's set up right now, the default values will take priority over the dotfile. Instead, it should be: defaults -> dotfile -> leiningen. If you put `merge-default-options` after `merge-file-options` it should work (but test it please).